src/transformers/models/splinter/tokenization_splinter.py

```diff
@@ -0,0 +1,164 @@
+
+    def build_inputs_with_special_tokens(
+        self, token_ids_0: List[int], token_ids_1: Optional[List[int]] = None
+    ) -> List[int]:
+        """
+        Build model inputs from a pair of sequence for question answering tasks by concatenating and adding special
+        tokens. A Splinter sequence has the following format:
+
+        - single sequence: ``[CLS] X [SEP]``
+        - pair of sequences for question answering: ``[CLS] question_tokens [QUESTION] . [SEP] context_tokens [SEP]``
+
+        Args:
+            token_ids_0 (:obj:`List[int]`):
+                The question token IDs if pad_on_right, else context tokens IDs
+            token_ids_1 (:obj:`List[int]`, `optional`):
+                The context token IDs if pad_on_right, else question token IDs
+
+        Returns:
+            :obj:`List[int]`: List of `input IDs <../glossary.html#input-ids>`__ with the appropriate special tokens.
+        """
+        if token_ids_1 is None:
+            return [self.cls_token_id] + token_ids_0 + [self.sep_token_id]
+
+        cls = [self.cls_token_id]
+        sep = [self.sep_token_id]
+        question_suffix = [self.question_token_id] + [self.convert_tokens_to_ids(".")]
+        if self.padding_side == "right":
+            # Input is question-then-context
+            return cls + token_ids_0 + question_suffix + sep + token_ids_1 + sep
+        else:
+            # Input is context-then-question
+            return cls + token_ids_0 + sep + token_ids_1 + question_suffix + sep
+
+    def create_token_type_ids_from_sequences(
+        self, token_ids_0: List[int], token_ids_1: Optional[List[int]] = None
+    ) -> List[int]:
+        """
+        Create the token type IDs corresponding to the sequences passed. `What are token type IDs?
+        <../glossary.html#token-type-ids>`__
+
+        Should be overridden in a subclass if the model has a special way of building those.
+
+        Args:
+            token_ids_0 (:obj:`List[int]`): The first tokenized sequence.
+            token_ids_1 (:obj:`List[int]`, `optional`): The second tokenized sequence.
+
+        Returns:
+            :obj:`List[int]`: The token type ids.
+        """
+        if token_ids_1 is None:
+            return [0] + [0] * len(token_ids_0) + [0]
+
+        if self.padding_side == "right":
+            # Input is question-then-context
+            return [0] + [0] * len(token_ids_0) + [0] + [0] + [0] + [1] * len(token_ids_1) + [1]
+        else:
+            # Input is context-then-question
+            return [0] + [0] * len(token_ids_0) + [0] + [1] * len(token_ids_1) + [1] + [1] + [1]
+
+    @property
+    def question_token_id(self) -> Optional[int]:
+        """
+        :obj:`Optional[int]`: Id of the question token in the vocabulary, used to condition the answer on a question
+        representation.
+
+        Returns :obj:`None` if the token has not been set.
+        """
+        if self._cls_token is None:
```

Comment:
Why check for the private cls token?

Reply:
I followed the conventions from ``tokenization_utils_base``.
I'll remove it.